### PR TITLE
feat: compare non exisiting user

### LINKED ISSUE #333: Feature: Compare for the non-existing user in the database
### Issue
Currently, compare feature doesn't allow to compare the users who are not in the database.

### Expected behavior
Any profile can be compared, irrespective of their presence in database

Please assign this issue to me

### PR DESCRIPTION
**Pull request by dhairyathedev**
Fixes #333

### CODE REVIEW

**Comment by ashutosh-rath02** [comment]
Thank you for submitting your pull request! 🙌 We'll review it as soon as possible. In the meantime, please ensure that your changes align with our [CONTRIBUTING.md](https://github.com/ashutosh-rath02/git-re/blob/main/CONTRIBUTING.md). If there are any specific instructions or feedback regarding your PR, we'll provide them here. Thanks again for your contribution! 😊

File: app/compare/page.tsx [hunk 1]
```diff
@@ -106,9 +106,24 @@ const Compare = () => {
...
-    const [rank] = await getIndividualUserRank(username);
+
+    let [rating, rankResult] = await Promise.all([
+      calculateRating(username),
+      getIndividualUserRank(username),
+    ]);
+    let rank = rankResult[0];
+
     const stats = await fetchUserStats(username);
+    console.log("Stats fetched:", stats);
```
Comment (ashutosh-rath02) [inline app/compare/page.tsx:117]: Remove this

Comment (dhairyathedev) [reply app/compare/page.tsx]: Done @ashutosh-rath02

**Comment by Swastik19Nit** [comment]
@dhairyathedev is it working??? i mean it is still not accepting user which is not in the leaderboard.

**Comment by ashutosh-rath02** [comment]
Removed labels as not fixed

**Comment by dhairyathedev** [comment]
> @dhairyathedev is it working??? i mean it is still not accepting user which is not in the leaderboard.

@Swastik19Nit https://github.com/ashutosh-rath02/git-re/issues/333#issuecomment-2198592984

**Comment by dhairyathedev** [comment]
Fixed by #342

**Review by ashutosh-rath02 — COMMENTED** [review]
LGTM

**Pull request merged**